src/Nix.hs

```diff
@@ -244,8 +250,10 @@ numberOfFetchers derivationContents =
   where
     countUp x = T.count x derivationContents
 
+-- Sum the number of things that look like fixed-output derivation hashes
 numberOfHashes :: Text -> Int
-numberOfHashes derivationContents = countUp "sha256 =" + countUp "sha256="
+numberOfHashes derivationContents =
+  sum $ map countUp ["sha256 =", "sha256=", "cargoSha256 =", "modSha256 ="]
   where
     countUp x = T.count x derivationContents
```

Comment:
It'd be nice to get something that isn't a heuristic here; without my update, the regular `version` updater will kick and do a half-baked job on the Rust upgrade, since it thinks it just has 1 hash.